baseapp/baseapp.go

```diff
@@ -199,8 +196,19 @@ func (app *BaseApp) Name() string {
 }
 
 // AppVersion returns the application's protocol version.
-func (app *BaseApp) AppVersion() uint64 {
-	return app.appVersion
+func (app *BaseApp) AppVersion(ctx context.Context) uint64 {
```

Comment:
Can we have this return an error? In the ABCI++ branch, all the ABCI methods return an error so I would prefer this not to panic.